src/components/description_list/description_list.tsx

```diff
@@ -119,8 +78,12 @@ export const EuiDescriptionList: FunctionComponent<
   }
 
   return (
-    <dl className={classes} {...rest}>
-      {childrenOrListItems}
-    </dl>
+    <EuiDescriptionListContext.Provider
+      value={{ type, compressed, textStyle, align }}
+    >
+      <dl className={classes} css={cssStyles} {...rest}>
```

Comment:
I would put the `data-type={type}` here and not on the child elements
```suggestion
      <dl className={classes} css={cssStyles} data-type={type} {...rest}>
```